cli/server_request_console.go

```diff
@@ -0,0 +1,109 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/net/websocket"
+)
+
+func newServerRequestConsoleCommand(cli *CLI) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "request-console [FLAGS] SERVER",
+		Short:                 "Request a WebSocket VNC console for a server",
+		Args:                  cobra.ExactArgs(1),
+		TraverseChildren:      true,
+		DisableFlagsInUseLine: true,
+		PreRunE:               cli.ensureToken,
+		RunE:                  cli.wrap(runServerRequestConsole),
+	}
+	cmd.Flags().BoolP("listen", "l", false, "Listen for a VNC client")
+	cmd.Flags().StringP("bind", "b", "localhost:5900", "Bind host/port for VNC port")
+	return cmd
+}
+
+func consoleProxy(bind, url string) error {
+	s, err := net.Listen("tcp", bind)
+	if err != nil {
+		return err
+	}
+	defer s.Close()
+
+	fmt.Printf("\nListening for VNC client on %v...\n", bind)
+
+	conn, err := s.Accept()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("VNC Client connect, establishing WebSocket connection...\n")
+	defer conn.Close()
+
+	ws, err := websocket.Dial(url, "binary", "https://console.hetzner.cloud/")
+	if err != nil {
+		return err
+	}
+	ws.PayloadType = websocket.BinaryFrame
+	fmt.Printf("Connected!\n")
+	defer ws.Close()
+
+	in := make(chan error)
```

Comment:
You can make the channel buffered (`make(chan error, 1)`) to avoid leaking goroutines.

Reply:
Thanks